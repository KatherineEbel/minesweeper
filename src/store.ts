import { configureStore } from '@reduxjs/toolkit'
import {reducer as minesweeper} from 'modules/MinesweeperRedux'

export const store = configureStore({
  reducer: {
    minesweeper
  }
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch