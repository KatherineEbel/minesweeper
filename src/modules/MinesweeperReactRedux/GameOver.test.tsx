import {render, screen} from '@testing-library/react'
import GameOverRedux from 'modules/MinesweeperReactRedux/GameOver'
import React from 'react'
import {Provider} from 'react-redux'
import {store} from 'store'

test('GameOver renders correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <GameOverRedux/>
    </Provider>
  )
  const gameOver = screen.getByText("ðŸ¤•")
  expect(gameOver.parentElement).toHaveStyle({transform: 'translate(-50%, -50%) scale(0)'})
  expect(asFragment()).toMatchSnapshot()
})