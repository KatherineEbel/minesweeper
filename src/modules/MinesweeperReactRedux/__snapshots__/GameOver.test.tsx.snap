// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-17cwb24"
  >
    <span
      class="hover:rotate-[360deg] hover:scale-150 transition-transform duration-500"
    >
      ðŸ¤•
    </span>
  </div>
</DocumentFragment>
`;
